File: solutions/csharp/atm/WithdrawalTransaction.cs
namespace ATM
{
    public class WithdrawalTransaction : Transaction
    {
        public WithdrawalTransaction(string transactionId, Account account, double amount)
            : base(transactionId, account, amount)
        {
        }

        public override void Execute()
        {
            account.Debit(amount);
        }
    }
}

File: solutions/golang/atm/withdrawal_transaction.go
package atm

type WithdrawalTransaction struct {
	BaseTransaction
}

func NewWithdrawalTransaction(transactionID string, account *Account, amount float64) *WithdrawalTransaction {
	return &WithdrawalTransaction{
		BaseTransaction: BaseTransaction{
			TransactionID: transactionID,
			Account:       account,
			Amount:        amount,
		},
	}
}

func (t *WithdrawalTransaction) Execute() error {
	return t.Account.Debit(t.Amount)
}
